tests/autobahn/test_autobahn.py

```diff
@@ -0,0 +1,131 @@
+import json
+import shutil
+import subprocess
+import sys
+from pathlib import Path
+from typing import Any, Dict, Generator, List
+
+import pytest
+from pytest import TempPathFactory
+from python_on_whales import docker
+
+
+@pytest.fixture(scope="session")
+def report_dir(tmp_path_factory: TempPathFactory) -> Path:
+    return tmp_path_factory.mktemp("reports")
+
+
+@pytest.fixture(scope="session", autouse=True)
+def build_autobahn_testsuite() -> Generator[None, None, None]:
+    if shutil.which("docker") is None:
```

Comment:
This check would only check if the docker client executable is present, not if the server side is running. We should really be checking whether it's possible to reach out to the service (which may be stopped on the machine).

Doesn't this python-on-whales have any API to verify this? If not, consider wrapping that `docker.build()` invocation in try-except and skipping on exception it emits when docker is stopped.

Reply:
python-on-whales will install docker if it's not installed, however, if it's not running it will throw a `DockerException`